libs/generic-view/ui/src/lib/interactive/form/input/checkbox-input.tsx

```diff
@@ -141,6 +214,28 @@ const Input = styled.input<{ $withError?: boolean }>`
         opacity: 1;
         visibility: visible;
       }
+
+      ${IndeterminateIcon} {
+        opacity: 0;
+        visibility: hidden;
+      }
+    }
+  }
+
+  &:indeterminate + ${Label} {
+    ${InputBox} {
+      border-color: ${({ theme }) => theme.color.grey1};
+      background-color: ${({ theme }) => theme.color.grey1};
+
+      ${CheckIcon} {
+        opacity: 0;
+        visibility: hidden;
+      }
+
+      ${IndeterminateIcon} {
+        opacity: 1;
+        visibility: visible;
+      }
     }
   }
 
```

Comment:
⚪ 
Perhaps it’s worth keeping the checkbox in its primitive form as a base component, and for encapsulation and reducing context complexity, we could create an extended version with additional states?